benchmarks/ShuffleBench.cu

```diff
@@ -40,6 +40,7 @@ static void benchmarkScatterGather(benchmark::State &state) {
   for (auto _ : state) {
     state.PauseTiming();
     temp_shuffler(tmp, in_container, seed);
+    checkCudaError(cudaDeviceSynchronize());
```

Comment:
I'd switch back to StdShuffle since we can't prove our shuffle is properly random. This may invalidate our benchmarks

Reply:
I am confident enough given that it passes all tests. Using std shuffle dramatically increases benchmark time.

Reply:
Do you have enough memory to precompute a few shuffles? The benchmarks are slow because gtest thinks the benchmark is twice as fast as it actually is so it runs more iterations. If we move that cost up front  it might help

Reply:
Id rather pay a time penalty now than invalidate our results and have to redo them later

Reply:
Probably not enough memory for the larger sizes. I can change it back to std shuffle when I get in the office next.